includes/User.php

```diff
@@ -2481,14 +2504,52 @@ public static function createLocalOptionName($option, $cityId = null, $sep = '-'
 	 * the option name is not normal. Should you have to use the separator, PLEASE MAKE
 	 * A PLAN TO NORMALIZE IT TO "-".
 	 *
-	 * @param string $preferenc
+	 * @param string $preference
 	 * @param string $value
 	 * @param int $cityId [optional, defaults to $wgCityId]
-	 * @param char $sep [optional, defaults to '-']
+	 * @param string $sep [optional, defaults to '-']
 	 *
 	 */
 	public function setLocalPreference($preference, $value, $cityId = null, $sep = '-') {
-		return $this->setOption(self::createLocalOptionName($preference, $cityId, $sep), $value);
+		$this->setGlobalPreference(self::localToGlobalPropertyName($preference, $cityId, $sep), $value);
+	}
+
+	/**
+	 * Set a global user preference.
+	 *
+	 * @param string $preference
+	 * @param string $value
+	 */
+	public function setGlobalPreference($preference, $value) {
+		global $wgPreferencesUseService;
+
+		if ($wgPreferencesUseService) {
+			$this->load();
+
+			if ($value) {
+				$value = str_replace("\r\n", "\n", $value);
+				$value = str_replace("\r", "\n", $value);
+				$value = str_replace("\n", " ", $value);
+			}
```

Comment:
Minor: This could be separated out into a helper function since it's at a lower level of abstraction than the rest of the function.